### PR TITLE
[Reward] Use refactoring code

### PR DESCRIPTION
**Pull request by laggu**
## Proposed changes

This is a last pr of reward refactoring
- use RewardDistributor and StakingManager instead of global function

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: consensus/istanbul/validator/weighted.go [hunk 1]
```diff
@@ -606,8 +606,8 @@ func (valSet *weightedCouncil) Refresh(hash common.Hash, blockNum uint64, chainI
 		}
 		weightedValidators[vIdx] = weightedVal
 
-		i := newStakingInfo.GetIndexByNodeId(weightedVal.address)
-		if i != reward.AddrNotFoundInCouncilNodes {
+		i, err := newStakingInfo.GetIndexByNodeId(weightedVal.address)
```
Comment (KimKyungup) [inline consensus/istanbul/validator/weighted.go]: How about this? :)
```suggestion
		idx, err := newStakingInfo.GetIndexByNodeId(weightedVal.address)
```

File: consensus/istanbul/backend/engine.go [hunk 2]
```diff
@@ -375,16 +374,19 @@ func (sb *backend) Finalize(chain consensus.ChainReader, header *types.Header, s
...
-			stakingInfo := reward.GetStakingInfoFromStakingCache(header.Number.Uint64())
+			stakingInfo := sb.GetStakingManager().GetStakingInfo(header.Number.Uint64())
 			if stakingInfo != nil {
 				kirAddr = stakingInfo.KIRAddr
 				pocAddr = stakingInfo.PoCAddr
 			}
 		}
 
-		reward.DistributeBlockReward(state, header, pocAddr, kirAddr, sb.governance)
+		err := sb.rewardDistributor.DistributeBlockReward(state, header, pocAddr, kirAddr)
```
Comment (jeonghonah) [inline consensus/istanbul/backend/engine.go]: minor comment:
```suggestion
		if err := sb.rewardDistributor.DistributeBlockReward(state, header, pocAddr, kirAddr); err != nil {
			return nil, err
		}
```

File: consensus/istanbul/backend/engine.go [hunk 3]
```diff
@@ -375,16 +374,19 @@ func (sb *backend) Finalize(chain consensus.ChainReader, header *types.Header, s
...
+		if err != nil {
+			return nil, err
+		}
 	} else {
-		err := reward.MintKLAY(state, header, sb.governance)
+		err := sb.rewardDistributor.MintKLAY(state, header)
```
Comment (jeonghonah) [inline consensus/istanbul/backend/engine.go]: same as above.

File: consensus/istanbul/validator/default.go [hunk 4]
```diff
@@ -395,7 +396,7 @@ func (valSet *defaultSet) F() int {
 
 func (valSet *defaultSet) Policy() istanbul.ProposerPolicy { return valSet.policy }
```
Comment (jeonghonah) [inline consensus/istanbul/validator/default.go]: Missing description. How about adding the description in the next PR?

File: consensus/istanbul/validator/weighted.go [hunk 5]
```diff
@@ -554,7 +554,7 @@ func (valSet *weightedCouncil) Policy() istanbul.ProposerPolicy { return valSet.
 // It return no error when weightedCouncil
```
Comment (jeonghonah) [inline consensus/istanbul/validator/weighted.go]:
```suggestion
// It returns no error when weightedCouncil
```

File: consensus/istanbul/validator/weighted.go [hunk 6]
```diff
@@ -554,7 +554,7 @@ func (valSet *weightedCouncil) Policy() istanbul.ProposerPolicy { return valSet.
...
 //   (1) already has up-do-date proposers
 //   (2) successfully calculated up-do-date proposers
```
Comment (jeonghonah) [inline consensus/istanbul/validator/weighted.go]:
```suggestion
//   (2) successfully calculates up-do-date proposers
```

File: consensus/istanbul/validator/weighted.go [hunk 7]
```diff
@@ -578,7 +578,7 @@ func (valSet *weightedCouncil) Refresh(hash common.Hash, blockNum uint64, chainI
 		return err
 	}
 
-	newStakingInfo := reward.GetStakingInfoFromStakingCache(blockNum + 1)
+	newStakingInfo := stakingManager.GetStakingInfo(blockNum + 1)
 
 	valSet.stakingInfo = newStakingInfo
 	if valSet.stakingInfo == nil {
```
Comment (jeonghonah) [inline consensus/istanbul/validator/weighted.go]: `valSet.stakingInfo == nil` means `newStakingInfo` is nil. Is it okay to access `newStakingInfo.UseGini` in the following code?

Comment (laggu) [reply consensus/istanbul/validator/weighted.go]: it returns err if valSet.stakingInfo == nil :)

File: consensus/istanbul/validator/default.go (hunk 4 above)
Comment (laggu) [reply consensus/istanbul/validator/default.go]: Sure. I'll make a new pr for adding descriptions.

File: consensus/istanbul/backend/engine.go (hunk 2 above)
Comment (laggu) [reply consensus/istanbul/backend/engine.go]: reflected

File: consensus/istanbul/backend/engine.go (hunk 3 above)
Comment (laggu) [reply consensus/istanbul/backend/engine.go]: reflected

**Pull request merged**